compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.cpp

```diff
@@ -361,9 +361,32 @@ LogicalResult ShuffleTensorOp::verifyRegions() {
 //===----------------------------------------------------------------------===//
 
 void ValueBarrierOp::build(OpBuilder &builder, OperationState &result,
-                           Value input) {
-  result.addOperands({input});
-  result.addTypes(input.getType());
+                           ValueRange input) {
+  result.addOperands(input);
+  result.addTypes(llvm::map_range(input, [](Value v) { return v.getType(); }));
+}
+
+LogicalResult ValueBarrierOp::verify() {
```

Comment:
Correct me if otherwise, but does Variadic allow zero operands and we need to verify that here?

Reply:
Yeah, it does allow zero operands. However, the operation is still perfectly valid if there are not operands/results, it would just be dead and eliminated by DCE. I still added a verifier check though, because it really doesn't matter.